modules/dataplex-autodq/main.tf

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  prefix = var.prefix == null ? "" : "${var.prefix}-"
+}
+
+resource "google_dataplex_datascan" "datascan" {
+  project      = var.project_id
+  location     = var.region
+  data_scan_id = "${local.prefix}${var.name}"
+  display_name = "${local.prefix}${var.name}"
+  description  = "Terraform Managed." # todo: parametrize
+  labels       = var.labels
+
+  data {
+    resource = var.data.resource
+    entity   = var.data.entity
+  }
+
+  execution_spec {
+    field = var.incremental_field
+    trigger {
+      dynamic "on_demand" {
+        for_each = try(var.execution_schedule) == null ? [""] : []
+        content {
+        }
+      }
+      dynamic "schedule" {
+        for_each = try(var.execution_schedule) != null ? [""] : []
```

Comment:
Most of these `try()` are not needed. For example, `var.execution_schedule` always succeeds even if it's value is null

Reply:
good point. These `try()` was needed when the variable was a nested object -- before refactoring to be an optional string. I will remove this.

Reply:
This is completed.